pytorch_lightning/trainer/connectors/training_trick_connector.py

```diff
@@ -55,4 +54,8 @@ def configure_accumulated_gradients(self, accumulate_grad_batches: Union[int, Di
             schedule = {0: accumulate_grad_batches}
             self.trainer.accumulation_scheduler = GradientAccumulationScheduler(schedule)
         else:
-            raise TypeError("Gradient accumulation supports only int and dict types")
+            raise MisconfigurationException(
+                f"Gradient accumulation supports only int and dict types, got {accumulate_grad_batches}"
+            )
+
+        self.trainer.accumulate_grad_batches = self.trainer.accumulation_scheduler.get_accumulate_grad_batches(0)
```

Comment:
I think what you are fixing here is this line, where you are already setting the value here instead of waiting for the `GradientAccumulationScheduler` to do it in the `on_train_epoch_start` hook. 
A bit more context on the PR description would help the reviewer next time :)

Reply:
is `trainer.accumulate_grad_batches` always expected to be an int? could we add a typehint for this if so?

tangentially related: what happens if someone passes both `accumulate_grad_batches=X` and a `GradientAccumulationScheduler` callback to the trainer? Should we detect this and raise a misconfiguration error in this case?

Reply:
@awaelchli updated the description.
thanks @ananthsub will check this :)